dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/ConsoleFormatTest.java

```diff
@@ -0,0 +1,479 @@
+
+    private static final String DEV_FORMAT_REGEX_PATTERN = "([A-Z]{3,}   )";
+    private static final String ISO_8601_REGEX_PATTERN = "(\\d{4})\\-(\\d{2})\\-(\\d{2})T(\\d{2})\\:(\\d{2})\\:(\\d{2})\\.(\\d{3})[+-](\\d{4})"; //ISO 8601 Format : yyyy-MM-dd'T'HH:mm:ss.SSSZ
+    private static final String INTERNAL_CLASSES_REGEXP = "at \\[internal classes\\]";
+
+    private static final String[] EXPECTED_FAILURES = { "SRVE0777E", "SRVE0315E", "CWWKG0032W", "CWWKG0075E" };
+
+    private static final int CONN_TIMEOUT = 10;
+
+    @Server(SERVER_NAME)
+    public static LibertyServer server;
+
+    @Server(SERVER_ENV_NAME)
+    public static LibertyServer serverEnv;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        ShrinkHelper.defaultDropinApp(server, "logger-servlet", "com.ibm.ws.logging.fat.logger.servlet");
+        ShrinkHelper.defaultDropinApp(server, "broken-servlet", "com.ibm.ws.logging.fat.broken.servlet");
+        server.startServer();
+
+        // Preserve the original server configuration
+        server.saveServerConfiguration();
+    }
+
+    @Before
+    public void setupTestStart() throws Exception {
+        if (server != null && !server.isStarted()) {
+            // Restore the original server configuration, with the default settings
+            server.restoreServerConfiguration();
+            server.startServer(true);
+        }
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        if (server != null && server.isStarted()) {
+            server.stopServer(EXPECTED_FAILURES);
+        }
+    }
+
+    /*
+     * This test dynamically changes the consoleFormat from dev to simple and vice-versa, and verifies if new messages are in the new correct format.
+     */
+    @Test
+    public void testDynamicSimpleFormat() throws Exception {
+        // Retrieve the consoleLogFile RemoteFile
+        RemoteFile consoleLogFile = server.getFileFromLibertyServerRoot(CONSOLE_LOG);
+
+        // Verify if the console logging format is in the default dev format
+        List<String> lines = server.findStringsInLogs(DEV_FORMAT_REGEX_PATTERN, consoleLogFile);
+        assertTrue("The console log is not in dev format.", lines.size() > 0);
+
+        // Set the consoleFormat="simple" and traceSpec=off in server.xml
+        setServerConfiguration(SIMPLE_FORMAT, false, false, consoleLogFile);
+
+        // Verify if the server was successfully updated
+        String line = server.waitForStringInLogUsingMark("CWWKG0017I", consoleLogFile);
+        Log.info(c, "testDynamicSimpleConsoleFormat", "The simple console formatted line : " + line);
+        assertNotNull("Message CWWKG0017I not appeared or appeared more than once ", line);
+
+        // Verify if the console.log file is in the simple format
+        assertFalse("The console.log file was not formatted to the simple format.", stringContainsPattern(line));
```

Comment:
Can you describe how `stringContainsPattern(line)` can be used to verify both simple format and dev format?

Reply:
So far, `stringContainsPattern(line)` method verifies only if the line contains the dev format pattern. I think it will be better to update the method name to something like `isStringinDevFormat(line)`? For the simple format, i think we can have a similar method, that matches the simple format format, instead of using the negative response of the `isStringinDevFormat(line)`?

Reply:
That will be a lot easier to understand.

Reply:
Updated.